app/scripts/controllers/transactions/index.js

```diff
@@ -331,6 +331,137 @@ export default class TransactionController extends EventEmitter {
     });
   }
 
+  // ====================================================================================================================================================
+
+  /**
+   * @param {number} txId
+   * @returns {TransactionMeta} the txMeta who matches the given id if none found
+   * for the network returns undefined
+   */
+  _getTransaction(txId) {
+    const { transactions } = this.store.getState();
+    return transactions[txId];
+  }
+
+  _validateTransaction(txId) {
+    return (
+      this._getTransaction(txId).status === TRANSACTION_STATUSES.UNAPPROVED
+    );
+  }
+
+  _updateTransaction(txParams, note) {
+    const normalizedTxParams = txUtils.normalizeTxParams(txParams);
+    const withAdditionalMeta = this.txStateManager.generateTxMeta({
+      txParams: { ...txParams, ...normalizedTxParams },
+      origin: 'metamask',
+    });
+    this.txStateManager.updateTransaction(withAdditionalMeta, note);
+  }
+
+  /**
+   * updates the gas fees of the transaction with id if the transaction state is unapproved
+   *
+   * @param {string} txId - transaction id
+   * @param {object} txGasFees - holds the gas fees parameters
+   * {
+   * gasLimit,
+   * gasPrice,
+   * maxPriorityFeePerGas,
+   * maxFeePerGas,
+   * estimateUsed
+   * }
+   */
+  updateTransactionGasFees(txId, txGasFees) {
```

Comment:
The documentation on each of these methods is good.

I think we could go a step further and strictly enforce that only those properties can be updated via this method. This could be done by destructuring the parameter:

```
updateTransactionGasFees(txId, {
   gasLimit,
   gasPrice,
   maxPriorityFeePerGas,
   maxFeePerGas,
   estimateUsed
}) {
```

That way, no additional properties could accidentally be included in the update